packages/platform-sdk-profiles/src/drivers/memory/profiles/profile.ts

```diff
@@ -274,9 +257,21 @@ export class Profile implements IProfile {
 
 		this.settings().fill(data.settings);
 
-		await this.restoreWallets(this, data.wallets);
+		await this.wallets().fill(data.wallets);
 
-		await this.contacts().fill(data.contacts);
+		this.contacts().fill(data.contacts);
+	}
+
+	/**
+	 * Sync the walles and contacts with their respective networks.
+	 *
+	 * @returns {Promise<void>}
+	 * @memberof Profile
+	 */
+	public async sync(): Promise<void> {
+		await this.wallets().syncAll();
```

Comment:
Can these two be done simultaneously? With a Promise.all()?

Reply:
Missed your comment but no because contacts re-use some data and instances from restored wallets.